tests/cupy_tests/core_tests/fusion_tests/fusion_utils.py

```diff
@@ -0,0 +1,124 @@
+        generate_inputs_name='generate_inputs',
+        generate_inputs_args=None,
+        check_array=None,
+        check_array_kwargs=None,
+        accept_error=()):
+    """Decorator for tests for ``cupy.fuse``.
+
+    This decorator checks
+
+    Args:
+        generate_input_args(tuple): argument tuple passed to
+            ``generate_input``. Defaults to ``()``.
+        check_array(function): testing function which compares
+            ``{numpy/cupy}.ndarray`` objects.
+            Defaults to ``testing.assert_allclose``.
+        check_array_kwargs(dict): keyword arguments passed to
+            ``check_array``. Defaults to ``{'rtol': 3e-3, 'atol': 3e-3}``.
+        accept_error(Exception or tuple of Exception):
+            Specify acceptable errors.
+    """
+    if generate_inputs_args is None:
+        generate_inputs_args = ()
+    if check_array is None:
+        check_array = testing.assert_allclose
+    if check_array_kwargs is None:
+        # TODO(imanishi): Relax tolerances only when comparing float16 arrays.
+        check_array_kwargs = {'rtol': 3e-3, 'atol': 3e-3}
+    if not isinstance(accept_error, (tuple, list)):
+        accept_error = (accept_error,)
+
+    def check(xp, actual, expected):
+        if expected is None:
+            assert actual is None
+
+        elif isinstance(expected, scalar_types + (numpy.ndarray,)):
+            assert isinstance(actual, scalar_types + (xp.ndarray,))
+            check_array(actual, expected, **check_array_kwargs)
+
+        elif isinstance(expected, (list, tuple)):
+            assert type(actual) == type(expected)
+            for item_actual, item_expected in zip(actual, expected):
+                check(xp, item_actual, item_expected)
+
+        elif isinstance(expected, dict):
+            assert isinstance(actual, dict)
+            for key, expected_value in expected:
+                actual_value = actual.pop(key)
+                check_array(xp, actual_value, expected_value)
+            assert len(actual) == 0
+
+        else:
+            assert False
+
+    def run(func, args, kwargs):
+        try:
+            ret = func(*args, **kwargs)
+            err = None
+        except Exception as e:
+            if not isinstance(e, accept_error):
+                raise
+            ret = None
+            err = e
+        return ret, err
+
+    def check_result(xp, actual, expected):
+        ret_a, err_a = actual
+        ret_e, err_e = expected
+
+        if err_e is None:
+            # Success
+            assert err_a is None
+            check(xp, ret_a, ret_e)
+
+    def deco(func):
+        def wrapper(self, **generate_inputs_kwargs):
+            generate_inputs = getattr(self, generate_inputs_name)
+
+            impl_np = func(self, numpy, **generate_inputs_kwargs)
+            impl_cp = func(self, cupy, **generate_inputs_kwargs)
+
+            # TODO(imanishi): Fix these workaround after `cupy.fuse`
+            # supports lambda function.
```

Comment:
It's not clear what are "these workaround".

Reply:
Enhanced the comment.